opt/jvmcommon.sh

```diff
@@ -5,7 +5,15 @@ calculate_java_memory_opts() {
   local memory_limit_file='/sys/fs/cgroup/memory/memory.limit_in_bytes'
 
   if [[ -f "${memory_limit_file}" ]]; then
-    case $(cat "${memory_limit_file}") in
+    local memory_limit
+    memory_limit=$(cat "${memory_limit_file}")
```

Comment:
In cgroups v1 (eg older versions of Docker), if no memory limit is set, this memory limits file has a bogus value of thousands of TB RAM. Whilst the old implementation here didn't filter out that case, it didn't need to due to the `case` statement providing some element of filtering (and the fallback case only setting `-XX:MaxRAMPercentage=80.0` and not actually using the memory value read from the limits file).

Now that the value from the limits file is used directly (to set `MaxRAM`), it seems you might want to filter out the bogus high value (that represents no limit being set)?

For Python I do that like so:
https://github.com/heroku/heroku-buildpack-python/blob/beb3328b618b3c2195d19690b483f031ba30ee9a/vendor/WEB_CONCURRENCY.sh#L34-L39

Reply:
That's a really good catch! Thanks a lot! Fixed in 4d36485

Reply:
@Malax @edmorley Docker sets `9223372036854771712` as `memory.limit_in_bytes`, that's (almost) 8 EiB (not quite `2^63-1` which would be `9223372036854775807`, discrepancy explained here: https://stackoverflow.com/questions/70332396).

There are AWS instances with 1 TB of RAM already ;) so maybe this limit check should be higher?

Reply:
I thought about that when implementing the Python solution. If someone is using an AWS instance with >= 1TB RAM and is also still using cgroups v1, then IMO they can (and probably should) set their own limits. Our automatic concurrency/... heuristics likely fall apart well before that point. (For example the 1.5 TB `r7i.48xlarge` instance type comes with 192 CPUs... for Python a gunicorn sync worker concurrency of ~192*2 probably isn't correct anyway).

Reply:
But I'd be fine with say using a limit of 10TB or something if you'd prefer. (I'm just not keen on specifying specific bogus high values, in case there are other scenarios where it's not precisely `9223372036854771712` but is still bogus.)

Reply:
Given that the main (and only officially supported) target machines are Heroku dynos I think a 1TiB limit is totally fine. As @edmorley said, this is just a way to filter out bogus values. I'm doubting the value of this discussion (apart from Ed's initial remark).